code/modules/mapping/map_template.dm

```diff
@@ -101,19 +137,36 @@
 	parsed.initTemplateBounds()
 
 	log_game("[name] loaded at [T.x],[T.y],[T.z]")
-	return bounds
+	on_map_loaded(T.z, parsed.bounds)
 
-/datum/map_template/proc/get_affected_turfs(turf/T, centered = FALSE)
-	var/turf/placement = T
-	if(centered)
-		var/turf/corner = locate(placement.x - round(width/2), placement.y - round(height/2), placement.z)
-		if(corner)
-			placement = corner
-	return block(placement, locate(placement.x+width-1, placement.y+height-1, placement.z))
+	return bounds
 
+//This, get_affected_turfs, and load() calculations for bounds/center can probably be optimized. Later.
+/datum/map_template/proc/annihilate_bounds(turf/origin, centered = FALSE, orientation = SOUTH)
+	var/deleted_atoms = 0
+	log_world("Annihilating objects in map loading location.")
+	var/list/turfs_to_clean = get_affected_turfs(origin, centered, orientation)
+	if(turfs_to_clean.len)
```

Comment:
You dont need the if, the for statements just wont do anything if it's empty
Also just put them both in the same loop
```
for(var/i in get_affected_turfs(origin, centered, orientation))
  var/turf/place = i
  for(var/k in place)
    ...
```

Also be careful how you clean up the turfs like this. We have a proc for emptying a turf, go check how that works and/or use it if possible

Reply:
ah woops I took this from polaris, converting to empty() next commit.